### PR TITLE
show first class object characteristics

### PR DESCRIPTION
**Pull request by udohjeremiah**
Since first class objects are being talked about here, it would be nice to provide an example of how a function is assigned to a variable, showing **"first class"** characteristics.

### CODE REVIEW

**Comment by ViralBShah** [comment]
Is it possible to make this a `jldoctest`?

**Comment by udohjeremiah** [comment]
I think so

File: doc/src/manual/functions.md [hunk 1]
```diff
@@ -257,7 +257,20 @@ This creates a function taking one argument `x` and returning the value of the p
...
+And because they're first class objects, we can assign them to a variable and then call them using
+the variable assigned to as a function:
+```jldoctest
+julia> A = x -> x^2 + 2x - 1
+#5 (generic function with 1 method)
+
+julia> A(10)
+119
+
+julia> A(-3)
```
Comment (KristofferC) [inline doc/src/manual/functions.md:269]: Does this extra call provide any more information than the first one did?

**Comment by KristofferC** [comment]
I tried to make this a bit terser but the description:

> they can be assigned to variables, and called using the standard function call syntax from the
variable they have been assigned to

is so clear that I doubt the example is really useful. This is also part of the manual and not directly documenting the Julia object so examples for everything is a bit disruptive. The julia syntax `->` alreday contains this information;:

```
help?> ->
search: ->

  x -> y

  Create an anonymous function mapping argument(s) x to the function body y.

  julia> f = x -> x^2 + 2x - 1
  #1 (generic function with 1 method)
  
  julia> f(2)
  7

  Anonymous functions can also be defined for multiple argumets.

  julia> g = (x,y) -> x^2 + y^2
  #2 (generic function with 1 method)
  
  julia> g(2,3)
  13
```

**Pull request closed without merging**